leafmap/common.py

```
@@ -16104,26 +16104,28 @@ def h5_to_gdf(
-            if columns is None or key in columns or key == lat or key == lon:
-                col_names.append(key)
-                col_val.append(value[:].tolist())
+    out_df = pd.concat(dfs, ignore_index=True) if dfs else pd.DataFrame()
 
-        df = pd.DataFrame(map(list, zip(*col_val)), columns=col_names)
-        out_df = pd.concat([out_df, df])
-        h5.close()
+    if lat not in out_df.columns or lon not in out_df.columns:
+        raise ValueError(
+            f"No {lat}/{lon} data found in dataset {dataset} of the input file(s)."
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)